controller/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+use anyhow::Result;
+use chrono::Utc;
+use dis_spawner::{
+    messages::agent::DroneStatusMessage,
+    messages::scheduler::{ScheduleRequest, ScheduleResponse},
+    nats::TypedNats,
+};
+use scheduler::Scheduler;
+use tokio::select;
+use tokio_stream::StreamExt;
+
+mod scheduler;
+
+pub async fn run_scheduler(nats: TypedNats) -> Result<()> {
+    let scheduler = Scheduler::default();
+    let mut spawn_request_sub = nats.subscribe(ScheduleRequest::subscribe_subject()).await?;
+
+    let mut status_sub = Box::pin(
+        nats.subscribe_jetstream(
+            &DroneStatusMessage::stream_name(),
+            &DroneStatusMessage::subscribe_subject(),
+        )
+        .await,
+    );
+
+    loop {
+        select! {
+            status_msg = status_sub.next() => {
+                let status_msg = status_msg.unwrap();
+
+                scheduler.update_status(Utc::now(), &status_msg);
```

Comment:
I wonder if there's a way to make this somewhat resistant to delays in message delivery 🤔  I suppose we could include a timestamp in the `DroneStatusMessage`, but then we'd be relying on the drone's clock and the controller's clock being relatively in sync.